src/lib/hierarchy-operations/index.js

```diff
@@ -0,0 +1,193 @@
+    } while (reportDocsBatch.length >= Backend.BATCH_SIZE);
+
+    return skip;
+  }
+
+  function reassignReports(reports, sourceId, destinationId, updatedReports) {
+    function reassignReportWithSubject(report, subjectId) {
+      let updated = false;
+      if (report[subjectId] === sourceId) {
+        report[subjectId] = destinationId;
+        updated = true;
+      }
+
+      if (report.fields[subjectId] === sourceId) {
+        report.fields[subjectId] = destinationId;
+        updated = true;
+      }
+
+      if (updated) {
+        const isAlreadyUpdated = !!updatedReports.find(updated => updated._id === report._id);
+        if (!isAlreadyUpdated) {
+          updatedReports.push(report);
+        }
+      }
+    }
+
+    for (const report of reports) {
+      const subjectIds = ['patient_id', 'patient_uuid', 'place_id', 'place_uuid'];
+      for (const subjectId of subjectIds) {
+        reassignReportWithSubject(report, subjectId);
+      }
+    }
+  }
+
+  function minifyLineageAndWriteToDisk(docs) {
+    docs.forEach(doc => {
+      lineageManipulation.minifyLineagesInDoc(doc);
+      JsDocs.writeDoc(options, doc);
+    });
+  }
+
+  function replaceLineageInReports(reportsCreatedByDescendants, replaceWith, startingFromId) {
+    return reportsCreatedByDescendants.reduce((agg, doc) => {
+      const replaceLineageOptions = {
+        lineageAttribute: 'contact',
+        replaceWith,
+        startingFromId,
+        merge: options.merge,
+      };
+
+      if (lineageManipulation.replaceLineage(doc, replaceLineageOptions)) {
+        agg.push(doc);
+      }
+      return agg;
+    }, []);
+  }
+
+  function replaceLineageInAncestors(descendantsAndSelf, ancestors) {
+    return ancestors.reduce((agg, ancestor) => {
+      let result = agg;
+      const primaryContact = descendantsAndSelf.find(descendant => ancestor.contact && descendant._id === ancestor.contact._id);
+      if (primaryContact) {
+        ancestor.contact = lineageManipulation.createLineageFromDoc(primaryContact);
+        result = [ancestor, ...result];
+      }
+    
+      return result;
+    }, []);
+  }
+
+  function replaceLineageInContacts(descendantsAndSelf, replaceWith, destinationId) {
+    function replaceForSingleContact(doc) {
+      const docIsDestination = doc._id === destinationId;
+      const startingFromId = options.merge || !docIsDestination ? destinationId : undefined;
+      const replaceLineageOptions = {
+        lineageAttribute: 'parent',
+        replaceWith,
+        startingFromId,
+        merge: options.merge,
+      };
+      const parentWasUpdated = lineageManipulation.replaceLineage(doc, replaceLineageOptions);
+
+      replaceLineageOptions.lineageAttribute = 'contact';
+      replaceLineageOptions.startingFromId = destinationId;
+      const contactWasUpdated = lineageManipulation.replaceLineage(doc, replaceLineageOptions);
+      const isUpdated = parentWasUpdated || contactWasUpdated;
+      if (isUpdated) {
```

Comment:
This is all because of SonarCube... basically moved a bunch of old code into functions to reduce the complexity and get it passing for this. 

I did a bad job. Who decided complexity of 5 was a good idea?  I basically find it too low to be practical.